### PR TITLE
Wait for transition before loading

### LINKED ISSUE #1406: Try to make loading use threading
to make the loading screen animation play

### LINKED ISSUE #3566: A black screen is now shown much longer than the loading screen when loading a save
probably due to the adjusted fade times, we should do something so that we would show the actual loading screen longer than the black screen. The only idea I really have is to make it so that the save load can't progress to the JSON deserialize step before the loading screen is fully visible. Because enabling multithreading for the loading will probably cause a ton of rare random crashes.

### PR DESCRIPTION
**Pull request by DrRadek**
**Brief Description of What This PR Does**

Waits for transition to finish before loading (Suggested by hhyyrylainen).
It's currently not possible to implement multi-threaded loading without major slowdown due to Godot 3 limitations (see https://github.com/godotengine/godot/issues/36793).

**Related Issues**

closes #3566

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [X] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
Thanks for the link about the threading thing, certainly very interesting. I did already know that Godot 4.0 makes background resource loading (more) possible.

I edited your PR description to mark this as actually not closing that other issue (which I updated with a new label and comment), but the more recent too long black screen issue #3566

File: src/saving/InProgressLoad.cs [hunk 1]
```diff
@@ -88,10 +88,14 @@ private void Step()
...
             {
+                // Wait for transition to finish.
+                // Workaround for #1406
+                if (TransitionManager.Instance.HasQueuedTransitions)
+                    break;
+
                 // Start suppressing loaded node deletion
                 TemporaryLoadedNodeDeleter.Instance.AddDeletionHold(Constants.DELETION_HOLD_LOAD);
 
-                // TODO: do this in a background thread if possible
```
Comment (hhyyrylainen) [inline src/saving/InProgressLoad.cs:94]: I'd keep this comment but reword it to 
```
// TODO: do this in a background thread once possible
// https://github.com/Revolutionary-Games/Thrive/issues/1406
```

File: src/saving/InProgressLoad.cs [hunk 2]
```diff
@@ -88,10 +88,14 @@ private void Step()
                 break;
             case State.ReadingData:
             {
+                // Wait for transition to finish.
```
Comment (hhyyrylainen) [inline src/saving/InProgressLoad.cs:91]: This class adds the transition on line 82, so shouldn't it be possible for this to also actually register the proper callback? I think checking if there are queued transitions is a dirty way to do it and I think we've managed to avoid using that elsewhere in the codebase.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/saving/InProgressLoad.cs (hunk 2 above)
Comment (DrRadek) [reply src/saving/InProgressLoad.cs]: Sure.

**Review by hhyyrylainen — APPROVED** [review]
Code is fine now and seems to work in game.
